src/map/Types.ts

```diff
@@ -19,6 +19,19 @@ export interface MarkerProps {
   key: string
 }
 
+export type AnimationFunction = (
+    {
+      prevMarkers,
+      markers,
+      setZoomType,
+      setConsolidatedMarkers
+    }: {
+      prevMarkers: ReactElement<MarkerProps>[];
+      markers: ReactElement<MarkerProps>[];
+      setZoomType: (zoomType: string | null) => void,
+      setConsolidatedMarkers: (markers: ReactElement<MarkerProps>[]) => void
+    }) => void;
```

Comment:
I wonder if instead of passing in setter functions (`setZoomType` and `setConsolidatedMarkers`), this function can return the associated values. This would make these functions a little easier to test.

```suggestion
export type AnimationFunction = (
    {
      prevMarkers,
      markers
    }: {
      prevMarkers: ReactElement<MarkerProps>[];
      markers: ReactElement<MarkerProps>[];
    }) => {
      zoomType: string | null;
      markers: ReactElement<MarkerProps>[];
    };
```